### PR TITLE
fix complete missing exit_code, when external commands' exit code stream is None

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Fixes: #7283

# User-Facing Changes
```
❯ (do -i { nu --testbin fail asdf } | complete)
╭───────────┬───╮
│ stdout    │   │
│ stderr    │   │
│ exit_code │ 0 │
╰───────────┴───╯
```
# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by WindSoilder** [comment]
I think something is wrong....

Because the exit code in `complete` is longer correct..
```
❯ do -i { git rev-parse --git-dir } | complete
╭───────────┬──────────────────────────────────────────────────────────────────╮
│ stdout    │                                                                  │
│ stderr    │ fatal: not a git repository (or any of the parent directories):  │
│           │ .git                                                             │
│           │                                                                  │
│ exit_code │ 128                                                              │
╰───────────┴──────────────────────────────────────────────────────────────────╯
                     
```

**Comment by theAkito** [comment]
> I think something is wrong....
> 
> Because the exit code in `complete` is longer correct..
> 
> Here is version 0.69.1, which works fine
> 
> ```
> ❯ do -i { git rev-parse --git-dir } | complete
> ╭───────────┬──────────────────────────────────────────────────────────────────╮
> │ stdout    │                                                                  │
> │ stderr    │ fatal: not a git repository (or any of the parent directories):  │
> │           │ .git                                                             │
> │           │                                                                  │
> │ exit_code │ 128                                                              │
> ╰───────────┴──────────────────────────────────────────────────────────────────╯
>                      
> ```

To give this more context:

I've been using this command for lots of versions. It always worked fine.

When I upgraded to 0.72.0, it stopped working and the whole issue began.

So, there must have been a very breaking change, between 0.71.0 & 0.72.0.

This claim is backed up by the commits to the `do` command. It was fiddled around with and now this does not work anymore.

**Comment by theAkito** [comment]
As explained [here](https://github.com/nushell/nushell/issues/7283#issuecomment-1331484870), the documentation says, my command is still correct, even after the commits to `do`.

Which is why the [reason for closing the related issue](https://github.com/nushell/nushell/issues/7283#issuecomment-1331494219) is definitely & utterly incorrect.

**Comment by WindSoilder** [comment]
Yeah, I know what you mean, I just mean that the implementation in my pr is not correct(exit_code value is wrong).  Here is my testing result:
```
❯ do -s { git rev-parse --git-dir } | complete
╭───────────┬──────────────────────────────────────────────────────────────────────╮
│ stdout    │                                                                      │
│ stderr    │ fatal: not a git repository (or any of the parent directories): .git │
│           │                                                                      │
│ exit_code │ 128                                                                  │
╰───────────┴──────────────────────────────────────────────────────────────────────╯
                                                                                         

❯ do -i { git rev-parse --git-dir } | complete
╭───────────┬──────────────────────────────────────────────────────────────────────╮
│ stdout    │                                                                      │
│ stderr    │ fatal: not a git repository (or any of the parent directories): .git │
│           │                                                                      │
│ exit_code │ 0                                                                    │
╰───────────┴──────────────────────────────────────────────────────────────────────╯
```

File: crates/nu-command/tests/commands/do_.rs [hunk 1]
```diff
@@ -98,6 +98,15 @@ fn ignore_error_should_work_for_external_command() {
     assert_eq!(actual.out, "post");
 }
 
+#[test]
+fn ignore_error_with_complete_should_have_exit_code() {
+    let actual = nu!(cwd: ".", pipeline(
+        r#"(do -i { nu --testbin fail asdf } | complete).exit_code"#
+    ));
+
+    assert_eq!(actual.out, "0");
```
Comment (theAkito) [inline crates/nu-command/tests/commands/do_.rs:107]: I'm no expert, but I want to help.

Shouldn't this be non-zero?

Comment (theAkito) [reply crates/nu-command/tests/commands/do_.rs:107]: Ah, I get it now. The new implementation of `-i` ignores all errors, so it should be zero.

**Comment by theAkito** [comment]
> ❯ do -i { git rev-parse --git-dir } | complete
> ╭───────────┬──────────────────────────────────────────────────────────────────────╮
> │ stdout    │                                                                      │
> │ stderr    │ fatal: not a git repository (or any of the parent directories): .git │
> │           │                                                                      │
> │ exit_code │ 0                                                                    │
> ╰───────────┴──────────────────────────────────────────────────────────────────────

According to the new implementation, this is correct, if I understood the new implementation correctly.

**Comment by theAkito** [comment]
```nu
  Run the block and ignore both shell and program errors
  > do -i { thisisnotarealcommand }

  Run the block and ignore shell errors
  > do -s { thisisnotarealcommand }
```

If it ignores "both shell and program errors", then the shell exit code should be zero.

Or that one is actually the program error.

File: crates/nu-command/tests/commands/do_.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/tests/commands/do_.rs:107]: On previous version, the value is 128, and I think the value might need to be 128

Comment (theAkito) [reply crates/nu-command/tests/commands/do_.rs:107]: Yes, thought they did stuff in https://github.com/nushell/nushell/pull/7122 and from what is said, especially in the documentation, it seems like `-i` is supposed to "ignore" either error, which would mean a zero, I guess. I don't know.

**Comment by WindSoilder** [comment]
> If it ignores "both shell and program errors", then the shell exit code should be zero.
Or that one is actually the program error.

Oh, thanks for your information, so current implementation is correct

File: crates/nu-command/tests/commands/do_.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/tests/commands/do_.rs:107]: Ok, so the exit code should be zero :-)

**Comment by WindSoilder** [comment]
Mark as draft for now, this may not needed due to comment:https://github.com/nushell/nushell/pull/7204#issuecomment-1337896214

**Pull request closed without merging**